### PR TITLE
feat(angular,angular-server): angular v14 minimum support

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue URL: Internal

## What is the new behavior?

Upgrades the `@ionic/angular` and `@ionic/angular-server` packages to target a minimum supported version of Angular 14.

**`@ionic/angular`**

- Removes dev-preview `environmentInjector` properties for standalone routing from `ion-router-outlet` and `ion-tabs`.
  - Ionic supports standalone component routing without any extra configuration
- `ComponentFactoryResolver` deprecated usage has been migrated to `createComponent` or `ViewContainerRef`, depending on the usage requirement
- Filled in types for router implementation
- Refactored dependency injection usage to use `inject` syntax (aligning with the Angular team's convention)

**Test Infrastructure**

- Removes the test infrastructure and test apps for Angular (`ng12`) and Angular 13 (`ng13`)
- Upgrades the Angular 15 test app to use v15 official dependencies

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: BREAKING.md [hunk 1]
```diff
@@ -289,8 +289,12 @@ Any references to the virtual scroll types from `@ionic/core` have been removed.
 
 <h4 id="version-7x-angular">Angular</h4>
 
+- Angular v14 is now required to use `@ionic/angular` and `@ionic/angular-server`. Upgrade your project to Angular v14 by following the [Angular v14 update guide](https://update.angular.io/?l=3&v=13.0-14.0).
```
Comment (liamdebeasi) [inline BREAKING.md:292]: We've been updating required version here too: https://github.com/ionic-team/ionic-framework/blob/feature-7.0/BREAKING.md#browser-and-platform-support

File: angular/test/apps/ng14/src/app/version-test/version-test-routing.module.ts [hunk 2]
```diff
@@ -3,12 +3,7 @@ import { RouterModule } from "@angular/router";
 
 @NgModule({
   imports: [
-    RouterModule.forChild([
-      {
-        path: 'standalone',
```
Comment (liamdebeasi) [inline angular/test/apps/ng14/src/app/version-test/version-test-routing.module.ts:8]: Do we still need standalone component tests? My understanding is we're not removing support, we're just removing the old environment injector prop

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

File: angular/test/apps/ng14/src/app/version-test/version-test-routing.module.ts (hunk 2 above)
Comment (sean-perkins) [reply angular/test/apps/ng14/src/app/version-test/version-test-routing.module.ts:8]: Standalone routing is now supported for all versions of Angular that we test. I just relocated the test (and updated the path) from being in each individual test app, to being in the "base" test app that all the test apps use.

**Review by liamdebeasi — APPROVED** [review]

**Comment by liamdebeasi** [comment]
Don't forget to update the commit description with the breaking change message too

**Pull request merged**